metadata-ingestion/tests/unit/test_teradata_source.py

```diff
@@ -1494,3 +1496,267 @@ def _create_mock_entry(
         entry.default_database = default_database
         entry.session_id = session_id or f"session_{query_id}"
         return entry
+
+
+class TestTeradataRegressionFixes:
+    """Test suite for regression fixes in Teradata source.
+
+    This class tests the fixes for two critical issues:
+    1. Time window defaults not being applied correctly
+    2. Missing incremental lineage support
+    """
+
+    def test_time_window_defaults_applied(self):
+        """Test that BaseTimeWindowConfig defaults are automatically applied.
+
+        This test ensures that the fix for the time window issue works correctly.
+        Previously, start_time and end_time were None before pydantic validation,
+        causing malformed SQL queries.
+        """
+        # Configuration without explicit time ranges (user's original recipe scenario)
+        config_dict = {
+            "username": "test_user",
+            "password": "test_password",
+            "host_port": "localhost:1025",
+            "include_table_lineage": True,
+            "include_usage_statistics": True,
+        }
```

Comment:
this is duplicated several times, what about having a fixture providing a basic config and then each tests does some basic udpates to the basic conf?